### PR TITLE
Assistant feature (code cleaning + new parameter)

### PR DESCRIPTION
**Pull request by laure-crochepierre**
### **Code cleaning : Only using the word "alarm" for ICAPS assistant track**

This PR : 
- [X]  adds warnings in observation, action and reward docstring  (taking into account issue #437)

- [x] renames variables containing the name "alert"  in BaseEnv

- [X] renames data_test folder "l2rpn_neurips_2020_track1_with_alert" to "l2rpn_neurips_2020_track1_with_alarm"

- [X] updates tests to look for this new test folder

_/!\ Warning : there is _**still**_ a file named "alerts_info.json" in "l2rpn_neurips_2020_track1_with_alarm" that we cannot change for now for compatibility reason (listed in issue #441)

### CODE REVIEW

**Comment by laure-crochepierre** [comment]
- [X] add parameter ALERT_TIME_HORIZON
- [X] Include docstring for this param 
- [X] Include tests

File: grid2op/tests/test_alert.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from grid2op.Episode import EpisodeData
+
+class TestAlert(unittest.TestCase):
+    """test the basic bahavior of the assistant alert feature"""
+
+    def test_init_default_param(self) -> None : 
+        self.env_nm = os.path.join(
+            PATH_DATA_TEST, "l2rpn_idf_2023_with_alert"
+        )
+        self.env = make(self.env_nm, test=True)
```
Comment (BDonnot) [inline grid2op/tests/test_alert.py:31]: can you add the `with warnings.catch_warnings() etc.` idioms that is present everywhere in the test when making an environment please ?

**Review by BDonnot — COMMENTED** [review]
A simple comments about the tests :-)

File: grid2op/tests/test_alert.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from grid2op.operator_attention import LinearAttentionBudget
+from grid2op import make
+from grid2op.Reward import RedispReward, _AlarmScore
+from grid2op.Exceptions import Grid2OpException
+from grid2op.Runner import Runner
+from grid2op.Environment import Environment
+from grid2op.Episode import EpisodeData
+
+class TestAlert(unittest.TestCase):
+    """test the basic bahavior of the assistant alert feature"""
```
[the remark this answers is not in the document]
Comment (laure-crochepierre) [reply grid2op/tests/test_alert.py:25]: Yes more tests will be added later. This PR is only about the name cleaning and the creation of the parameter. 
Thus, this test in only about the "parameter". In order to use the budget correctly, I need to create new observation variables, they will be added in a next PR.

**Review by BDonnot — APPROVED** [review]

**Pull request merged**